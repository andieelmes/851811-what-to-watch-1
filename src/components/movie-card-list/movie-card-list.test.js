import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardList from 'components/movie-card-list/movie-card-list.jsx';

const movies = [
  {
    id: 1,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
  },
  {
    id: 2,
    img: `img/bohemian-rhapsody.jpg`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    title: `Bohemian Rhapsody`,
  },
];

describe(`Movie card list component`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(
          <MovieCardList
            movies={movies}
          />, {createNodeMock: (el) => el})
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
