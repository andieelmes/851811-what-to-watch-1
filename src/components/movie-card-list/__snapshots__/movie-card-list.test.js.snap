// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card list component should render correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    data-movie-id={1}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      style={null}
    >
      <img
        alt="Fantastic Beasts: The Crimes of Grindelwald"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        style={null}
        width="280"
      />
      <video
        controls={true}
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    data-movie-id={2}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      style={null}
    >
      <img
        alt="Bohemian Rhapsody"
        height="175"
        src="img/bohemian-rhapsody.jpg"
        style={null}
        width="280"
      />
      <video
        controls={true}
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
</div>
`;
